src/main/java/io/github/nucleuspowered/nucleus/modules/ticket/data/TicketDataCache.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.modules.ticket.data;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import io.github.nucleuspowered.nucleus.api.nucleusdata.Ticket;
+import org.apache.commons.lang3.tuple.Triple;
+
+import java.sql.*;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+public class TicketDataCache {
+    private TicketDataManager ticketDataManager;
+    private Map<Integer, TicketData> cachedTickets = Maps.newHashMap(); //TODO Actually cache retrieved data and serve it.
+
+    public TicketDataCache(TicketDataManager ticketDataManager) {
+        this.ticketDataManager = ticketDataManager;
+    }
+
+    public CompletableFuture<List<Ticket>> lookupTicket(List<Triple<String, String, String>> arguments) throws SQLException {
+        CompletableFuture<List<Ticket>> future = new CompletableFuture<>();
```

Comment:
Create the future, then do everything else off the thread - so use `Task.asyc` to wrap the rest of this code and return the future. Otherwise, you'll still be blocking the thread that calls this until it completes.